File: mobile/lib/models/inmueble.dart
class Inmueble {
  final String id;
  final String siniestroId;
  final String tipo;
  final String? tipoInmuebleId;
  final int numeroNiveles;
  final String tipoUnidad;
  final bool esPadre;
  final String? padreId;
  final String identificador;
  final String estadoAfectacion;
  final String observaciones;
  final bool sincronizado;

  Inmueble({
    required this.id,
    required this.siniestroId,
    this.tipo = '',
    this.tipoInmuebleId,
    this.numeroNiveles = 1,
    this.tipoUnidad = '',
    this.esPadre = false,
    this.padreId,
    this.identificador = '',
    this.estadoAfectacion = 'sin_daños',
    this.observaciones = '',
    this.sincronizado = false,
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'siniestroId': siniestroId,
        'tipo': tipo,
        'tipoInmuebleId': tipoInmuebleId,
        'numeroNiveles': numeroNiveles,
        'tipoUnidad': tipoUnidad,
        'esPadre': esPadre ? 1 : 0,
        'padreId': padreId,
        'identificador': identificador,
        'estadoAfectacion': estadoAfectacion,
        'observaciones': observaciones,
        'sincronizado': sincronizado ? 1 : 0,
      };

  factory Inmueble.fromMap(Map<String, dynamic> map) => Inmueble(
        id: map['id'] as String,
        siniestroId: map['siniestroId'] as String,
        tipo: map['tipo'] as String? ?? '',
        tipoInmuebleId: map['tipoInmuebleId'] as String?,
        numeroNiveles: map['numeroNiveles'] as int? ?? 1,
        tipoUnidad: map['tipoUnidad'] as String? ?? '',
        esPadre: (map['esPadre'] as int? ?? 0) == 1,
        padreId: map['padreId'] as String?,
        identificador: map['identificador'] as String? ?? '',
        estadoAfectacion: map['estadoAfectacion'] as String? ?? 'sin_daños',
        observaciones: map['observaciones'] as String? ?? '',
        sincronizado: (map['sincronizado'] as int? ?? 0) == 1,
      );

  Map<String, dynamic> toJson() => {
        'tipo': tipo,
        'tipo_inmueble_ref': tipoInmuebleId,
        'numero_niveles': numeroNiveles,
        'tipo_unidad': tipoUnidad,
        'es_padre': esPadre,
        'identificador': identificador,
        'estado_afectacion': estadoAfectacion,
        'observaciones': observaciones,
      };
}

File: backend/models/Inmueble.js
const mongoose = require('mongoose');

const inmuebleSchema = new mongoose.Schema({
  siniestro: { type: mongoose.Schema.Types.ObjectId, ref: 'Siniestro', required: true },
  tipo: { type: String, default: '' },
  tipo_inmueble_ref: { type: mongoose.Schema.Types.ObjectId, ref: 'TipoInmueble', default: null },
  numero_niveles: { type: Number, default: 1 },
  tipo_unidad: { type: String, default: '' },
  es_padre: { type: Boolean, default: false },
  padre: { type: mongoose.Schema.Types.ObjectId, ref: 'Inmueble', default: null },
  identificador: { type: String, default: '' },
  estado_afectacion: { type: String, enum: ['critico', 'moderado', 'sin_daños'], default: 'sin_daños' },
  observaciones: { type: String, default: '' },
  sincronizado: { type: Boolean, default: true },
  creado_en: { type: Date, default: Date.now },
  actualizado_en: { type: Date, default: Date.now },
});

inmuebleSchema.pre('save', function (next) {
  this.actualizado_en = Date.now();
  next();
});

module.exports = mongoose.model('Inmueble', inmuebleSchema);
